aten/src/ATen/native/ChanelShuffle.cpp

```diff
@@ -0,0 +1,30 @@
+#include <ATen/native/TensorTransformations.h>
+
+#include <ATen/NativeFunctions.h>
+#include <c10/util/Exception.h>
+
+#include <algorithm>
+#include <vector>
+
+namespace at {
+namespace native {
+
+Tensor channel_shuffle(const Tensor& self, int64_t groups) {
+  AT_ASSERTM(self.dim() == 4,
+             "channel_shuffle expects 4D input, but got input with sizes ",self.sizes());
```

Comment:
Maybe relax this constraint to `dim > 2`?  Should be achievable by

1. First calculate num_pixels as the product of all dimensions other than 0 and 1.
1. `input_reshaped = self.reshape({b, groups, oc, num_pixels})`
1. Change `reshape({b, c, h, w})` to `rehsape(self.shape())`

Reply:
Sounds reasonable.